### PR TITLE
Update Services to return empty response in case of error.

### PR DESCRIPTION
**Pull request by josh8551021**
Originally, when there was an internal error on the server, the default behavior would be to send back a nil response and an error. However, this behavior is not correct. Instead, we should send back an empty response.

This PR also addresses #21, where only a single response would be returned for a `CreateDirectory` request. The issue stemmed from the fact that a lock on an internal data structure was not being released before a method returned.

### CODE REVIEW

File: src/core/cluster/cluster_service.go [hunk 1]
```diff
@@ -115,7 +117,7 @@ func (c *ClusterServer) UpdateFileContentsCS(ctx context.Context,
 	request *hootpb.UpdateFileContentsCSRequest) (*hootpb.UpdateFileContentsCSResponse, error) {
 	file_id, err := uuid.FromBytes(request.FileId.Value)
 	if err != nil {
-		return nil, ErrInvalidId
+		return nil, status.Error(codes.InvalidArgument, ErrInvalidId.Error())
```
Comment (Landmaster) [inline src/core/cluster/cluster_service.go:120]: Is keeping the 1st return value nil intentional here, or was this an oversight?

File: src/core/cluster/cluster_service.go [hunk 2]
```diff
@@ -128,7 +130,7 @@ func (c *ClusterServer) UpdateFileContentsCS(ctx context.Context,
 func (c *ClusterServer) MoveObjectCS(ctx context.Context, request *hootpb.MoveObjectCSRequest) (*hootpb.MoveObjectCSResponse, error) {
 	obj_id, err := uuid.FromBytes(request.CurrObjectId.Value)
 	if err != nil {
-		return nil, ErrInvalidId
+		return nil, status.Error(codes.InvalidArgument, ErrInvalidId.Error())
```
Comment (Landmaster) [inline src/core/cluster/cluster_service.go:133]: Same here.

File: src/core/cluster/cluster_service.go (hunk 1 above)
Comment (josh8551021) [reply src/core/cluster/cluster_service.go:120]: That was an oversight. I'll fix this.

**Review by chathamabate — APPROVED** [review]
LGTM

File: src/core/cluster/cluster_service.go (hunk 2 above)
Comment (josh8551021) [reply src/core/cluster/cluster_service.go]: Done.

File: src/core/cluster/cluster_service.go (hunk 1 above)
Comment (josh8551021) [reply src/core/cluster/cluster_service.go]: Done.

**Review by Landmaster — APPROVED** [review]
should be fine now

**Pull request merged**